### PR TITLE
[colotensor] Bert 

### PR DESCRIPTION
**Pull request by Wesley-Jzy**

### CODE REVIEW

File: language/bert/colotensor/utils.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+import os
+
+import colossalai.utils as utils
+import torch
+from torch.distributed import all_reduce, get_rank, get_world_size, is_initialized
+from colossalai.utils import colo_set_process_memory_fraction, get_current_device, MultiTimer
+from colossalai.core import global_context as gpc
+from concurrent.futures import ThreadPoolExecutor
+
+class AsyncMemoryMonitor:
```
Comment (FrankLeeeee) [inline language/bert/colotensor/utils.py:13]: Why is this needed?

File: language/bert/colotensor/dataset/wikitext.py [hunk 2]
```diff
@@ -48,26 +48,23 @@ def seed_worker():
...
-    test_data = DataLoader(
-        tokenized_dataset["validation"],
-        sampler=test_sampler,
-        drop_last=True,
-        collate_fn=default_data_collator,
-        worker_init_fn=seed_worker,
-        batch_size=batch_size,
-        pin_memory=True,
-    )
+    data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=True, mlm_probability=0.15)
```
Comment (FrankLeeeee) [inline language/bert/colotensor/dataset/wikitext.py:51]: This does not include NSP task of BERT pretraining. I have written a script to deal with the nsp dataset and do bert pretraining. The links are 
- https://github.com/hpcaitech/ColossalAI-Examples/tree/main/language/bert/preprocessing
- https://github.com/hpcaitech/ColossalAI-Examples/tree/main/language/bert/zero

File: language/bert/colotensor/dataset/wikitext.py [hunk 3]
```diff
@@ -48,26 +48,23 @@ def seed_worker():
...
+    train_sampler = DistributedSampler(tokenized_dataset['train'], shuffle=True) if world_size > 1 else None
+    train_data = DataLoader(tokenized_dataset['train'],
```
Comment (FrankLeeeee) [inline language/bert/colotensor/dataset/wikitext.py:53]: I notice that your dataset is wikitext, but it should be wikipedia dataset.

File: language/bert/colotensor/dataset/wikitext.py (hunk 2 above)
Comment (Wesley-Jzy) [reply language/bert/colotensor/dataset/wikitext.py:51]: ok. I will change the model.

File: language/bert/colotensor/utils.py (hunk 1 above)
Comment (Wesley-Jzy) [reply language/bert/colotensor/utils.py:13]: Because engine is not used, I use it to show statistics.

**Comment by binmakeswell** [comment]
Hi, @Wesley-Jzy Is this PR still active?
